contracts/access/AccessControlled.sol

```diff
@@ -32,55 +29,26 @@ abstract contract AccessControlled {
         _;
     }
 
-    /// @notice Ensures that the caller is a registered IP account.
-    /// @dev Modifier that checks if the msg.sender is a registered IP account.
-    /// modules can use this modifier to check if the caller is a registered IP account.
-    /// so that enforce only registered IP Account can call the functions.
-    modifier onlyIpAccount() {
-        if (!IP_ACCOUNT_REGISTRY.isIpAccount(msg.sender)) {
-            revert Errors.AccessControlled__CallerIsNotIpAccount(msg.sender);
-        }
-        _;
-    }
-
     /// @dev Constructor
     /// @param accessController The address of the AccessController contract.
-    /// @param ipAccountRegistry The address of the IPAccountRegistry contract.
     /// @custom:oz-upgrades-unsafe-allow constructor
-    constructor(address accessController, address ipAccountRegistry) {
+    constructor(address accessController) {
         if (accessController == address(0)) revert Errors.AccessControlled__ZeroAddress();
-        if (ipAccountRegistry == address(0)) revert Errors.AccessControlled__ZeroAddress();
         ACCESS_CONTROLLER = IAccessController(accessController);
-        IP_ACCOUNT_REGISTRY = IIPAccountRegistry(ipAccountRegistry);
     }
 
     /// @dev Internal function to verify if the caller (msg.sender) has the required permission to execute
-    /// the function on provided ipAccount.
+    /// the function on provided ipAccount. Reverts if the msg.sender does not have permission.
     /// @param ipAccount The address of the IP account to verify.
     function _verifyPermission(address ipAccount) internal view {
-        if (!IP_ACCOUNT_REGISTRY.isIpAccount(ipAccount)) {
-            revert Errors.AccessControlled__NotIpAccount(ipAccount);
-        }
-
-        if (msg.sender != ipAccount) {
-            // revert if the msg.sender does not have permission
-            ACCESS_CONTROLLER.checkPermission(ipAccount, msg.sender, address(this), msg.sig);
-        }
+        ACCESS_CONTROLLER.checkPermission(ipAccount, msg.sender, address(this), msg.sig);
     }
```

Comment:
Should NOT remove this logic. if it is called by IPAccount then don't need to call AccessController to check permission.

Reply:
Moved the logic to inside the AccessController. AccessControlled should not introduce any additional access check

Reply:
Unless there's a difference between making a custom caller check outside `checkPermission` and making it inside `checkPermission`

Reply:
In the current design, IPAccount has already performed `checkPermission` with AccessController before executing the functions. hence no need to call `checkPermission` with AccessController again. avoid extra duplicate checkpermission.

Reply:
What about calling a function directly that uses `verifyPermission`?

There are two cases:
1. Call through IPAccount, which performs `checkPermission` in `_isValidSigner` before executing the encoded call 
2. Call module directly

In (2), if we don't go through the usual `checkPermission`, then we're not fully checking ACL.

Reply:
we shall reserve the check of calling through IPAccount, even though it seems not widely used yet, because we don't have much app being built yet.

Reply:
let's not touch code too much at this stage

Reply:
Ok, understood. Want to note that we will have a duplicate check called on all direct calls to modules: checking that IP Account is a valid IP Account twice. The ACL is not straightforward in design atm.

Reply:
duplicated check seems like a potential issue, can you open a github issue so that we can discuss it over there? @jdubpark